From 283879d01773175d8c69acaaaa1c9faf8b173200 Mon Sep 17 00:00:00 2001
From: Ahmad Fatoum <a.fatoum@pengutronix.de>
Date: Wed, 27 Mar 2019 14:31:28 +0100
Subject: [PATCH 4/4] libusbgx: fix --without-libconfig build against
 glibc-2.28

Commit 45c14ef4d5 ("libusbgx: fix build with glibc-2.28 since
<sys/sysmacros.h> is no more included by <sys/types.h>")
fixed this issue for the (default) --with-libconfig configuration,
but building --without-libconfig still throws a linker error
because makedev was assumed to be an external symbol in
usbg_common.c.

Include <sys/sysmacros.h> there to address this.

Signed-off-by: Ahmad Fatoum <a.fatoum@pengutronix.de>
Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>
---
 src/usbg_common.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/usbg_common.c b/src/usbg_common.c
index f8822fc..6f12a38 100644
--- a/src/usbg_common.c
+++ b/src/usbg_common.c
@@ -15,6 +15,7 @@
 
 #include <errno.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <sys/types.h>
 #include <malloc.h>
 #include <stdio.h>
-- 
2.22.0

